docs/Gopkg.toml.md

```diff
@@ -0,0 +1,105 @@
+# Gopkg.toml
+
+## `required`
+required lists a set of packages (not projects) that must be included in Gopkg.lock. This list is merged with the set of packages imported by the current project. Use it when your project needs a package it doesn't explicitly import - including "main" packages.
+ ```toml
+required = ["github.com/user/thing/cmd/thing"]
+```
+
+## `ignored`
+ignored lists a set of packages (not projects) that are ignored when dep statically analyzes source code. Ignored packages can be in this project, or in a dependency.
+```toml
+ignored = ["github.com/user/project/badpkg"]
+```
+
+## `[metadata]`
+`metadata` can exist at the root as well as under `constraint` and `override`. 
+At the root `metadata` defines `metadata` about the project that could be used by other independent systems. 
+
+`metadata` under `constraint` or `override`  defines `metadata` about the dependency or override that could be used by other independent systems. The `metadata` defined here will be ignored by dep.
```

Comment:
As long as we are consistent, I think it's okay.

@darkowlzz What do you think?